### ISSUE TITLE
some confusions about custom DERP server

### ISSUE THREAD

**Issue by koalang** [tailscale/tailscale#9787]
I deploy a custom derp server and I used the "verify-clients" flag. The docs say 

> [Optional: Restricting client access to your DERP node ](https://tailscale.com/kb/1118/custom-derp-servers/#optional-restricting-client-access-to-your-derp-node)
Anyone that knows the IP address of your DERP node could add it to their DERP map and route their tailnet traffic through your DERP node. To allow only your tailnet traffic through your DERP node, run tailscaled on the same machine as your DERP node, and start derper with the --verify-clients flag:

> sudo derper --hostname=your-hostname.com --verify-clients

Shall I run run tailscaled and login in the same server?

**Comment by DentonGentry** [comment]
Yes, to use `--verify-clients` you run a `tailscaled` on the same server and join the tailnet which you wish to restrict clients to.

> 2023/10/13 04:19:33 http: TLS handshake error from 167.94.138.49:53798: cert mismatch with hostname: ""
> Why did it happen?

There is something wrong with the certificate. I assume this is not a certificate from LetsEncrypt. You need to ensure that the certificate matches the domain name of the derp server you're using it with, and that the root cert which signed the DERP certificate is populated in all of the rest of your client machines.

> why it showed not in set of peers

Please fix the certificate problem first, this is possibly a second order effect.

**Comment by bradfitz** [comment]
I'm going to close this for now, as it looks like the question has been answered. (and there's no bug to fix or other action to be taken)

**Comment by koalang** [comment]
Thanks for your reply.
> Yes, to use --verify-clients you run a tailscaled on the same server and join the tailnet which you wish to restrict clients to.
It means that if the derp sever used --verify-clients flag that will be a client machine in my account?

> There is something wrong with the certificate. I assume this is not a certificate from LetsEncrypt. You need to ensure that the certificate matches the domain name of the derp server you're using it with

I have deployed two derps, using TrusaAsia ECC DV and ZeroSSL RSA wildcard cert. It happens to all of them. But the amazing thing is that one of them I've tried to relay successfully by using `tailscale status`

> and that the root cert which signed the DERP certificate is populated in all of the rest of your client machines.

Shall I merge the root cert to my derp server cert, Is this working? I'm not sure if other client machines have corresponding root certificate. For android and windows clients, I usually use a browser to access it, the address bar shows up fine and I can view the full chain of certificates; for linux clients, I use curl -I -v my.derp.serve to ensure that I can connect.

May be I need to deploy lets encrypt certificates to try and see if the `cert mismatch with hostname: ""` situations appeared.

**Comment by DentonGentry** [comment]
> I have deployed two derps, using TrusaAsia ECC DV and ZeroSSL RSA wildcard cert.

You'll need to be responsible for the provisioning of TLS certificates for the node you're trying to operate, we cannot help with general system IT. The derper process implements LetsEncrypt automation, for anything else you need to install the appropriate root certificates.
